packages/driver/src/cy/net-stubbing/add-command.ts

```diff
@@ -0,0 +1,283 @@
+import _ from 'lodash'
+
+import {
+  RouteHandler,
+  RouteMatcherOptions,
+  RouteMatcher,
+  StaticResponse,
+  HttpRequestInterceptor,
+  STRING_MATCHER_FIELDS,
+  DICT_STRING_MATCHER_FIELDS,
+  AnnotatedRouteMatcherOptions,
+  AnnotatedStringMatcher,
+  NetEventFrames,
+  StringMatcher,
+  NumberMatcher,
+} from '@packages/net-stubbing/lib/types'
+import {
+  validateStaticResponse,
+  getBackendStaticResponse,
+  hasStaticResponseKeys,
+} from './static-response-utils'
+import { registerEvents } from './events'
+import $errUtils from '../../cypress/error_utils'
+
+/**
+ * Get all STRING_MATCHER_FIELDS paths plus any extra fields the user has added within
+ * DICT_STRING_MATCHER_FIELDS objects
+ */
+function getAllStringMatcherFields (options: RouteMatcherOptions): string[] {
+  return STRING_MATCHER_FIELDS
+  .concat(
+    // add the nested DictStringMatcher values to the list of fields to annotate
+    _.flatten(
+      _.filter(
+        DICT_STRING_MATCHER_FIELDS.map((field) => {
+          const value = options[field]
+
+          if (value) {
+            return _.keys(value).map((key) => {
+              return `${field}.${key}`
+            })
+          }
+
+          return ''
+        }),
+      ),
+    ),
+  )
+}
+
+/**
+ * Annotate non-primitive types so that they can be passed to the backend and re-hydrated.
+ */
+function annotateMatcherOptionsTypes (options: RouteMatcherOptions) {
+  const ret: AnnotatedRouteMatcherOptions = {}
+
+  getAllStringMatcherFields(options).forEach((field) => {
+    const value = _.get(options, field)
+
+    if (value) {
+      _.set(ret, field, {
+        type: (isRegExp(value)) ? 'regex' : 'glob',
+        value: value.toString(),
+      } as AnnotatedStringMatcher)
+    }
+  })
+
+  const noAnnotationRequiredFields = ['https', 'port', 'webSocket']
+
+  _.extend(ret, _.pick(options, noAnnotationRequiredFields))
+
+  return ret
+}
+
+function getUniqueId () {
+  return `${Number(new Date()).toString()}-${_.uniqueId()}`
+}
+
+function isHttpRequestInterceptor (obj): obj is HttpRequestInterceptor {
+  return typeof obj === 'function'
+}
+
+function isRegExp (obj): obj is RegExp {
+  return obj && (obj instanceof RegExp || obj.__proto__ === RegExp.prototype || obj.__proto__.constructor.name === 'RegExp')
+}
+
+function isStringMatcher (obj): obj is StringMatcher {
+  return isRegExp(obj) || _.isString(obj)
+}
+
+function isNumberMatcher (obj): obj is NumberMatcher {
+  return Array.isArray(obj) ? _.every(obj, _.isNumber) : _.isNumber(obj)
+}
+
+function validateRouteMatcherOptions (routeMatcher: RouteMatcherOptions): void {
+  if (_.isEmpty(routeMatcher)) {
```

Comment:
I see now that this is used in a `try/catch` below and throws a cypress error if caught, so you can ignore the above comment. 

However, I don't really like using `try/catch` as a control flow method, because it's throwing an error that we don't actually want to be **the** error. The real intention of this function is to _validate_ and _provide a message_. So I prefer the pattern of this return something like `{ isValid: true }` or `{ isValid: false, message: 'https must be a boolean' }` and using an `if` for the control flow below.

Reply:
good point, this felt icky. fixed this here & a related issue in validateStaticResponse